src/dft/backends/rocfft/commit.cpp

```diff
@@ -0,0 +1,426 @@
+                rocfft_status_success) {
+                throw mkl::exception("dft/backends/rocfft", __FUNCTION__,
+                                     "Failed to get forward work buffer size.");
+            }
+            if (work_buf_size != 0) {
+                void* work_buf;
+                if (hipMalloc(&work_buf, work_buf_size) != hipSuccess) {
+                    throw mkl::exception("dft/backends/rocfft", __FUNCTION__,
+                                         "Failed to get allocate forward work buffer.");
+                }
+                handles[0].buffer = work_buf;
+                if (rocfft_execution_info_set_work_buffer(fwd_info, work_buf, work_buf_size) !=
+                    rocfft_status_success) {
+                    throw mkl::exception("dft/backends/rocfft", __FUNCTION__,
+                                         "Failed to set forward work buffer.");
+                }
+            }
+        }
+
+        if (valid_backward) {
+            auto res =
+                rocfft_plan_description_set_data_layout(plan_desc, bwd_array_ty, fwd_array_ty,
+                                                        in_offsets.data(), // in offsets
+                                                        out_offsets.data(), // out offsets
+                                                        dimensions,
+                                                        in_strides.data(), //in strides
+                                                        bwd_distance, // in distance
+                                                        dimensions,
+                                                        out_strides.data(), // out strides
+                                                        fwd_distance // out distance
+                );
+            if (res != rocfft_status_success) {
+                throw mkl::exception("dft/backends/rocfft", __FUNCTION__,
+                                     "Failed to set backward data layout.");
+            }
+
+            if (rocfft_plan_description_set_scale_factor(plan_desc, config_values.bwd_scale) !=
+                rocfft_status_success) {
+                throw mkl::exception("dft/backends/rocfft", __FUNCTION__,
+                                     "Failed to set backward scale factor.");
+            }
+
+            rocfft_plan bwd_plan;
+            res = rocfft_plan_create(&bwd_plan, placement, bwd_type, precision, dimensions,
+                                     lengths.data(), number_of_transforms, plan_desc);
+            if (res != rocfft_status_success) {
+                throw mkl::exception("dft/backends/rocfft", __FUNCTION__,
+                                     "Failed to create backward rocFFT plan.");
+            }
+            handles[1].plan = bwd_plan;
+
+            rocfft_execution_info bwd_info;
+            if (rocfft_execution_info_create(&bwd_info) != rocfft_status_success) {
```

Comment:
how will the syncronization for the creation happen with multiple descriptors existing and multiple execution happening as well ?

Reply:
Sorry, I'm not sure I understand the question.
The different descriptors should not interact, apart from `rocfft_setup`, but that is handled with the singleton.
The descriptor doesn't have define behavior for what happens when you can it while an execution is ongoing.

Reply:
I see, with your comment below its seems like the plan_creation and execution(per stream sync) are synchronous, so this should be fine.